buku

```diff
@@ -553,35 +567,33 @@ class BukuDb:
         Returns
         -------
         int
-            DB index, or -1 if URL not found in DB.
+            DB index, or None if URL not found in DB.
         """
 
-        self.cur.execute('SELECT id FROM bookmarks WHERE URL = ? LIMIT 1', (url,))
-        resultset = self.cur.fetchall()
-        return resultset[0][0] if resultset else -1
+        row = self._fetch_first('SELECT * FROM bookmarks WHERE url = ?', url)
+        return row and row.id
 
     def get_max_id(self) -> int:
         """Fetch the ID of the last record.
 
         Returns
         -------
         int
-            ID of the record if any record exists, else -1.
+            ID of the record if any record exists, else None.
```

Comment:
Same answer as well – when there's no value to return, `None` is what should be used instead of an arbitrary number.

Reply:
-1 is not arbitrary. It's documented when it is being returned. Please revert.

Reply:
Just because it's documented doesn't make it any less arbitrary. `-1` is a number and it means a negative amount or index (which isn't what it's being used for), `None` is a value meant specifically to represent a lack of a thing.